nasl/nasl_ssh.c

```diff
@@ -1896,3 +1897,63 @@ nasl_ssh_shell_close (lex_ctxt *lexic)
 
   return NULL;
 }
+
+/*
+ * NASL SFTP
+ */
+
+/**
+ * @brief Check if SFTP is enabled.
+ * @naslfn{sftp_enabled_check}
+ *
+ * @nasluparam
+ *
+ * - An ssh session id.
+ *
+ * @naslret An integer: 0 on success, -1 (SSH_ERROR) on Channel request
+ * subsystem failure. Greather than 0 means an error during SFTP init.
+ *
+ * @param[in] lexic Lexical context of NASL interpreter.
+ */
+tree_cell *
+nasl_sftp_enabled_check (lex_ctxt *lexic)
+{
+  int tbl_slot, session_id;
+  tree_cell *retc;
+  sftp_session sftp;
+  ssh_session session;
+  int rc;
+
+  session_id = get_int_var_by_num (lexic, 0, -1);
+  if (!verify_session_id (session_id, "sftp_enabled_check", &tbl_slot, lexic))
+    return NULL;
```

Comment:
Is there a special distinction between errors via retc and NULL? If not we can just return retc with a new negative value. Else return NULL should at least be documented in the doxygen.

Reply:
Added docstring. I prefer to differentiate the errors. In this case is a nasl error (e.g. bad session id, not present in session table)